### PR TITLE
Add nss recipe

### PR DESCRIPTION
**Pull request by marcelotrevisani**
Add nss

https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS

Based on the work of @jakirkham , #536

### CODE REVIEW

File: recipes/nss/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  skip: true  # [win or osx]
+
+requirements:
+  build:
+    - autoconf
+    - cmake
+    - nspr
+    - sqlite
+    - zlib
+    - libgcc
```
Comment (jakirkham) [inline recipes/nss/meta.yaml]: We shouldn't need `libgcc` here as we are just using the system copy of the compiler run time currently.

File: recipes/nss/build.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    NS_USE_GCC=1 \
+    NSDISTMODE='copy' \
+    all latest
+
+cd ../dist
+FOLDER=$(<latest)
+cd $FOLDER
+
+cp -rL bin "${PREFIX}"
+cp -rL lib "${PREFIX}"
```
Comment (jakirkham) [inline recipes/nss/build.sh]: Is there no `make install`?

Comment (marcelotrevisani) [reply recipes/nss/build.sh]: I tried that but it didn't work.. I search in the documentation and they tell you to manually install that. That was the reason why I did that "by hand".

"The build system of NSS originated from Netscape's build system, which predated the "configure; make; make test; make install" sequence that we're familiar with now. Our makefiles also have an "install" target, but it has a different meaning: our "install" means installing the headers, libraries, and programs in the appropriate directories under mozilla/dist. "

https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/Reference/Building_and_installing_NSS/Installation_guide

But for me that is still fuzzy, however if you know a better solution for that, please let me know and I will try! :)

**Comment by marcelotrevisani** [comment]
Hi @jakirkham , thanks for review the recipe! Any new suggestion please let me know!  :)
Waiting for the comments of @isuruf

File: recipes/nss/build.sh (hunk 2 above)
Comment (marcelotrevisani) [reply recipes/nss/build.sh]: Sure, I will take a look! Thanks!

File: recipes/nss/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  number: 0
+  skip: true  # [win or osx]
+
+requirements:
+  build:
+    - autoconf
+    - cmake
+    - nspr
+    - sqlite
+    - zlib
```
[the remark this answers is not in the document]
Comment (marcelotrevisani) [reply recipes/nss/meta.yaml]: What is the argument ```config``` which I have to pass to your script?

File: recipes/nss/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  commands:
+    - test -f ${PREFIX}/lib/libnspr4${SHLIB_EXT}         # [unix]
+    - test -f ${PREFIX}/lib/libplds4${SHLIB_EXT}         # [unix]
+    - test -f ${PREFIX}/lib/libplc4${SHLIB_EXT}          # [unix]
+    - test -f ${PREFIX}/lib/libsoftokn3${SHLIB_EXT}      # [unix]
+    - test -f ${PREFIX}/lib/libsoftokn3.chk              # [unix]
+    - test -f ${PREFIX}/lib/libnss3${SHLIB_EXT}          # [unix]
+    - test -f ${PREFIX}/lib/libsmime3${SHLIB_EXT}        # [unix]
+    - test -f ${PREFIX}/lib/libssl3${SHLIB_EXT}          # [unix]
+    - test -f ${PREFIX}/lib/libnssckbi${SHLIB_EXT}       # [unix]
```
[the remark this answers is not in the document]
Comment (marcelotrevisani) [reply recipes/nss/meta.yaml]: I checked for the libraries which are described in 

https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS/Reference/Building_and_installing_NSS/Installation_guide

Do you think we need to test others than that? If yes, please let me know which one to add into the recipe! :)

[the remark this answers is not in the document]
Comment (jakirkham) [reply recipes/nss/meta.yaml]: Nope, that's fine. Sounds like there are no others.

File: recipes/nss/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,63 @@
+{% set version = "3.36.1" %}
+{% set version_us = version.replace(".", "_") %}
+
+package:
+  name: nss
+  version: {{ version }}
+
+source:
+  patches: nss-3.36.1-standalone-1.patch
```
Comment (jakirkham) [inline recipes/nss/meta.yaml]: Could you please move this below the source and format like this? Also filling out a comment explaining the patch would help.

```yaml
  patches:
    # <add description here>
    - nss-3.36.1-standalone-1.patch
```

Comment (marcelotrevisani) [reply recipes/nss/meta.yaml]: Got it, going to change

**Comment by marcelotrevisani** [comment]
@jakirkham , Do you have any new suggestions ? Can we merge that? :)

**Comment by jakirkham** [comment]
Nope, looks great. Let's merge it. :)

**Comment by jakirkham** [comment]
Thanks @marcelotrevisani :)

**Pull request merged**